src/argilla_sdk/records/_dataset_records.py

```diff
@@ -382,11 +396,106 @@ def _normalize_batch_size(self, batch_size: int, records_length, max_value: int)
+        metadata: Dict[str, MetadataValue] = {}
+
+        schema = self.__dataset.schema
+
+        for attribute, value in data.items():
+            schema_item = schema.get(attribute)
+            attribute_type = None
+            sub_attribute = None
+
+            # Map source data keys using the mapping
+            if mapping and attribute in mapping:
+                attribute_mapping = mapping.get(attribute)
+                attribute_mapping = attribute_mapping.split(".")
+                attribute = attribute_mapping[0]
+                schema_item = schema.get(attribute)
+                if len(attribute_mapping) > 1:
+                    attribute_type = attribute_mapping[1]
+                if len(attribute_mapping) > 2:
+                    sub_attribute = attribute_mapping[2]
+            elif schema_item is mapping is None and attribute != "id":
+                warnings.warn(
+                    message=f"""Record attribute {attribute} is not in the schema so skipping. 
+                        Define a mapping to map source data fields to Argilla Fields, Questions, and ids
+                        """
+                )
+                continue
+
+            if attribute == "id":
+                record_id = value
+                continue
+
+            # Add suggestion values to the suggestions
+            if attribute_type == "suggestion":
+                if sub_attribute in ["score", "agent"]:
+                    suggestion_values[attribute][sub_attribute] = value
+
+                elif sub_attribute is None:
+                    suggestion_values[attribute].update(
+                        {"value": value, "question_name": attribute, "question_id": schema_item.id}
+                    )
+                else:
+                    warnings.warn(
+                        message=f"Record attribute {sub_attribute} is not a valid suggestion sub_attribute so skipping."
+                    )
+                continue
+
+            # Assign the value to question, field, or response based on schema item
+            if isinstance(schema_item, TextField):
+                fields[attribute] = value
+            elif isinstance(schema_item, QuestionType) and attribute_type == "response":
+                responses.append(Response(question_name=attribute, value=value, user_id=user_id))
+            elif isinstance(schema_item, QuestionType) and attribute_type is None:
+                suggestion_values[attribute].update(
+                    {"value": value, "question_name": attribute, "question_id": schema_item.id}
+                )
+            elif isinstance(schema_item, VectorField):
+                vectors.append(Vector(name=attribute, values=value))
+            elif isinstance(schema_item, MetadataType):
+                metadata[attribute] = value
+            else:
+                warnings.warn(message=f"""Record attribute {attribute} is not in the schema or mapping so skipping.""")
+                continue
```

Comment:
Do we want to skip it entirely? Not sure if it is better to skip or to break the loop? Otherwise we might miss one random record from the total.